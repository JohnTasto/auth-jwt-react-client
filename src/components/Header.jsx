import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { actions as authActions } from '../ducks/auth'


export class Header extends Component {
  renderLinks() {
    if (this.props.authenticated) {
      return (
        <li className="nav-item">
          <Link className="nav-link" to="/" onClick={this.props.signOut}>Sign Out</Link>
        </li>
      )
    } else {  // eslint-disable-line no-else-return
      return ([
        <li className="nav-item" key={1}>
          <Link className="nav-link" to="/signin">Sign In</Link>
        </li>,
        <li className="nav-item" key={2}>
          <Link className="nav-link" to="/signup">Sign Up</Link>
        </li>,
      ])
    }
  }

  render() {
    return (
      <nav className="navbar navbar-light">
        <Link to="/" className="navbar-brand">Redux Auth</Link>
        <ul className="nav navbar-nav">
          <li className="nav-item">
            <Link className="nav-link" to="/feature">Feature</Link>
          </li>
          {this.renderLinks()}
        </ul>
      </nav>
    )
  }
}

export default connect(
  state => ({
    authenticated: state.auth.authenticated,
  }),
  authActions,
  (...props) =>
    Object.assign({}, ...props),
)(Header)
