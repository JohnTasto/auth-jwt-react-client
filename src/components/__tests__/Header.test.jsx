/* eslint-env jest */
/* eslint-disable comma-dangle */

import React from 'react'
import { StaticRouter, Route } from 'react-router-dom'
import { Provider } from 'react-redux'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import createStore from '../../store'
import { Header } from '../Header'


describe('<Header />', () => {
  test('Renders signin/signup links if not authenticated', () => {
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  test('Renders signout link if authenticated', () => {
    const wrapper = shallow(<Header authenticated />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  describe('Integration', () => {
    let wrapper
    let props

    beforeEach(() => {
      props = {
        authenticated: true,
        signOut: jest.fn(),
      }
      wrapper = mount(
        <Provider store={createStore()}>
          <StaticRouter location="/" context={{}} >
            <Route
              path="/"
              render={routeProps => (<Header {...routeProps} {...props} />)}
            />
          </StaticRouter>
        </Provider>
      )
    })

    test('Calls action signOut when signout link is clicked', () => {
      wrapper.find('li a[href="/"]').simulate('click')

      expect(props.signOut).toHaveBeenCalled()
    })
  })
})
