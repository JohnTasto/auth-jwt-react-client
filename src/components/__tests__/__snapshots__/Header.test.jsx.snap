// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Renders signin/signup links if not authenticated 1`] = `
<nav
  className="navbar navbar-light"
>
  <Link
    className="navbar-brand"
    replace={false}
    to="/"
  >
    Redux Auth
  </Link>
  <ul
    className="nav navbar-nav"
  >
    <li
      className="nav-item"
    >
      <Link
        className="nav-link"
        replace={false}
        to="/feature"
      >
        Feature
      </Link>
    </li>
    <li
      className="nav-item"
    >
      <Link
        className="nav-link"
        replace={false}
        to="/signin"
      >
        Sign In
      </Link>
    </li>
    <li
      className="nav-item"
    >
      <Link
        className="nav-link"
        replace={false}
        to="/signup"
      >
        Sign Up
      </Link>
    </li>
  </ul>
</nav>
`;

exports[`<Header /> Renders signout link if authenticated 1`] = `
<nav
  className="navbar navbar-light"
>
  <Link
    className="navbar-brand"
    replace={false}
    to="/"
  >
    Redux Auth
  </Link>
  <ul
    className="nav navbar-nav"
  >
    <li
      className="nav-item"
    >
      <Link
        className="nav-link"
        replace={false}
        to="/feature"
      >
        Feature
      </Link>
    </li>
    <li
      className="nav-item"
    >
      <Link
        className="nav-link"
        replace={false}
        to="/"
      >
        Sign Out
      </Link>
    </li>
  </ul>
</nav>
`;
